test/gateway/eip20_gateway/stake.js

```diff
@@ -234,7 +234,7 @@ contract('EIP20Gateway.stake() ', (accounts) => {
     await mockToken.approve(gateway.address, stakeAmount, { from: stakerAddress });
     await baseToken.approve(gateway.address, bountyAmount, { from: stakerAddress });
 
-    nonce = new BN(2);
+    nonce = new BN(1);
```

Comment:
hardcoded on purpose? Why not "get nonce"?

Reply:
It is an existing code. 
I am in favor of using hardcoded values, as we know what should be the nonce.
Any change in the contract logic will break the test.

Why do you think that it should get from "get nonce" ?